src/components/data-management/SamplesTable.jsx

```diff
@@ -221,19 +222,14 @@ const SamplesTable = forwardRef((props, ref) => {
   };
 
   useEffect(() => {
-    if (!samples[activeProject.samples[0]]) {
+    if (activeProject.samples.length === 0) {
       setTableData([]);
       return;
     }
 
-    // Set table data
     const newData = activeProject.samples.map((sampleUuid, idx) => {
-      // upload problems sometimes lead to partial updates and incosistent states
-      // in this situation it's possible that the sampleUuid does not exist
-      // this a temporary fix so that the whole UI doesn't crash preventing the
-      // user from removing the dataset or uploading another one.
-      // this situation should be included into the SamplesTable.jsx tests
-      const sampleFiles = samples[sampleUuid]?.files || {};
+      const sampleData = samples[sampleUuid];
```

Comment:
for old projects with partially uploaded samples, wouldn't this fail?

Reply:
No, `sampleData` would be assigned `undefined`. The line that causes the error and the bug in the previous case was line 257 in the old file, where we try to access the property `name` of `undefined`